zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java

```diff
@@ -374,6 +376,34 @@ private HttpRequestMessage buildZuulHttpRequest(
         return request;
     }
 
+    private String parsePath(String uri) {
+        String path;
+        // relative uri
+        if (uri.startsWith("/")) {
+            path = uri;
+        } else {
+            Matcher m = URL_REGEX.matcher(uri);
+
+            // absolute uri
+            if (m.matches()) {
+                String match = m.group(3);
+                path = match == null ? uri : match;
```

Comment:
This doesn't read great. Consider making the assignment explicit on each branch

Reply:
Fixed.